Sources/BraveWallet/Crypto/Stores/SendTokenStore.swift

```diff
@@ -179,8 +181,35 @@ public class SendTokenStore: ObservableObject {
     self.ipfsApi = ipfsApi
     self.assetManager = userAssetManager
 
-    self.keyringService.add(self)
-    self.rpcService.add(self)
+    self.keyringServiceObserver = KeyringServiceObserver(
+      keyringService: keyringService,
+      _selectedWalletAccountChanged: { [weak self] _ in
+        guard let self else { return }
+        self.selectedSendTokenBalance = nil
+        self.addressError = nil
+        self.update() // `selectedSendTokenBalance` needs updated for new account
+        self.validateSendAddress() // `sendAddress` may equal selected account address
+      }
+    )
+    self.rpcServiceObserver = JsonRpcServiceObserver(
+      rpcService: rpcService,
+      _chainChangedEvent: { [weak self] _, coin, _ in
+        guard let self else { return }
+        self.selectedSendToken = nil
+        self.selectedSendTokenBalance = nil
+        if coin != .eth { // if changed to ethereum coin network, address is still valid
+          // offchain resolve is for ENS-only, for other coin types the address is invalid
+          self.isOffchainResolveRequired = false
+        }
+        self.update() // `selectedSendToken` & `selectedSendTokenBalance` need updated for new chain
+        self.validateSendAddress() // `sendAddress` may no longer be valid if coin type changed
+      }
+    )
+  }
+  
+  func tearDown() {
+    keyringServiceObserver = nil
+    rpcServiceObserver = nil
```

Comment:
`SelectAccountTokenStore` should conform to `WalletSubStore`, and we should call `tearDown()` on it  here 🙂. That way it remains active until Send is closed / `tearDown()` call on `SendTokenStore`